tests/integration/test_integration.py

```diff
@@ -114,60 +103,48 @@ async def post_deployment_tasks(self, namespace: str) -> str:
         generated_token = vault.generate_token(ttl="5m")
         return generated_token
 
-    async def test_given_no_config_when_deploy_then_status_is_waiting(  # noqa: E501
-        self, ops_test, charm, cleanup
+    @pytest.mark.abort_on_fail
+    async def test_given_no_config_when_deploy_then_status_is_blocked(  # noqa: E501
+        self, ops_test: OpsTest, build_and_deploy
     ):
-        await self.deploy_charm(ops_test, charm)
-
-        await ops_test.model.wait_for_idle(apps=[APPLICATION_NAME], status="blocked", timeout=1000)
+        await ops_test.model.wait_for_idle(apps=[APPLICATION_NAME], status="blocked", timeout=1000)  # type: ignore[union-attr]
 
+    @pytest.mark.abort_on_fail
     async def test_given_no_config_when_post_deployment_tasks_and_authorise_charm_then_status_is_active(  # noqa: E501
-        self, ops_test, charm, cleanup
+        self, ops_test: OpsTest, build_and_deploy, post_deployment_tasks
     ):
-        """This test follows the README.MD deployment and post-deployment tasks.
+        """This test follows the README.md deployment and post-deployment tasks.
 
         Args:
             ops_test: Ops test Framework
-            charm: Charm path
-
-        Returns:
-            None
+            build_and_deploy: Pytest fixture
+            post_deployment_tasks: Pytest fixture
         """
-        await self.deploy_charm(ops_test, charm)
-        vault_unit = ops_test.model.units["vault-k8s/0"]
-        await ops_test.model.wait_for_idle(apps=[APPLICATION_NAME], status="blocked", timeout=1000)
-
-        vault_token = await self.post_deployment_tasks(namespace=ops_test.model_name)
+        vault_unit = ops_test.model.units["vault-k8s/0"]  # type: ignore[union-attr]
 
+        vault_token = post_deployment_tasks
```

Comment:
Why remove the `await` here?

Reply:
This is a fixture, I think we don't need to await.

Reply:
This comment it's deprecated, it's no longer a fixture.